server/src/db.rs

```diff
@@ -573,14 +573,31 @@ impl Db {
 
                     // We should only report persisted chunks or chunks that are currently being persisted, because the
                     // preserved catalog does not care about purely in-mem chunks.
+
+                    // Capture ID of to be created chunk for the in-progress compacting OS chunks
+                    let (to_be_created_chunk_id, compacting_os) =
+                        match chunk.in_lifecycle_compacting_object_store() {
+                            Some(id) => (Some(id), true),
+                            _ => (None, false),
+                        };
+
                     if matches!(chunk.stage(), ChunkStage::Persisted { .. })
                         || chunk.is_in_lifecycle(ChunkLifecycleAction::Persisting)
+                        || compacting_os
                     {
                         affected_persisted_chunks.push(ChunkAddrWithoutDatabase {
                             table_name: Arc::clone(&chunk.addr().table_name),
                             partition_key: Arc::clone(&chunk.addr().partition_key),
                             chunk_id: chunk.addr().chunk_id,
                         });
+
+                        if compacting_os {
```

Comment:
```suggestion
                        if let Some(to_be_created) = to_be_created_chunk_id {
```

Should remove the need for the boolean and the `unwrap`